dojo-chess/src/models.cairo

```diff
@@ -0,0 +1,53 @@
+use starknet::ContractAddress;
+
+#[derive(Model, Drop, Serde)]
+struct Square {
```

Comment:
This is much more work but I think it would help to understand the example if we have a type `Piece` instead of `Square` and an implementation of utils functions such as:

```cairo
#[derive(Model, Drop, Serde)]
struct Piece {
#[key]
    game_id: u32,
    #[key]
    x: u32,
    #[key]
    y: u32,
    type: PieceType,
}
```

And the trait to implement:

```cairo
trait PieceTrait {
    fn is_mine(self: Piece) -> bool;
    fn is_out_of_board(next_position: (u32, u32)) -> bool;
    fn is_right_piece_move(self: Piece, curr_position: (u32, u32), next_position: (u32, u32)) -> bool;
}
```

Also `is_correct_turn` could be inside a `GameTrait`.

Reply:
Nothing mandatory, only my perspective ofc.

Reply:
Great idea. Gonna work on it